From 6b158b6c39d7a39a1a0cf976a2ad029e8b4e896c Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 8 Oct 2021 13:29:47 +0000
Subject: [PATCH 0976/1000] MANTIS 0061647 extension: 0061647:
 Conveyor-Component: new mode for conveyor: maintenance, forceBag, off
 platform.idl V5.7 add ConveyorState       CON_MAINTENANCE,      
 CON_FORCEBAG,       CON_OFF

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1678 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 23 +++++++++++--------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 06fa6de3..04638ed1 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "5.6";
+    const string interface_version = "5.7";
 
     /** Platform interface for the device components. Via this interface the device
         components communicate with the platform manager internally. */
@@ -96,9 +96,9 @@ module PlatformManagement
                                           the linked components can be found.
                                           (Index counting starts with 0) */
         string realComponentNameBase; /**< Real component name(e.g. ATB_PRNT_BP, ATB_PRNT_TK) */
-        string unitId;                /**< UnitId for logging */  
+        string unitId;                /**< UnitId for logging */
         LoggingClass unitClass;       /**< UnitClass for logging */
-        string unitIdBase;            /**< UnitId for logging */  
+        string unitIdBase;            /**< UnitId for logging */
     };
 
     typedef sequence<mEnvironmentComponent> mEnvironmentComponents;
@@ -190,11 +190,11 @@ module PlatformManagement
             the mediaOffRoller components will playback the stored events
         */
         code resetMediaRecord(in mRealComponents mediaOffRoller);
-        
+
         code enableDevice();
         code disableDevice();
         code switchEnableDevice(out boolean on);
-        
+
         code checkBindingConflict();
     };
 
@@ -329,10 +329,13 @@ module PlatformManagement
     enum ConveyorState
     {
       CON_READY,
-      CON_MANUAL_MODE, 
+      CON_MANUAL_MODE,
       CON_SWITCHED_OFF,
       CON_PARKED,
-      CON_ERROR
+      CON_ERROR,
+      CON_MAINTENANCE,
+      CON_FORCEBAG,
+      CON_OFF
     };
 
     interface AppStateListener
@@ -450,14 +453,14 @@ module PlatformManagement
         void logIllumination(in string unitID, in IlluminationEvent illuminationEvent);
 
         void logCounterValue(in string unitId, in CounterType counter, in long currentLevel, in long warningLevel, in long status);
-        
+
         code enableDisplay();
         code disableDisplay();
         code switchEnableDisplay(out boolean on);
-        
+
         code registerDevice(in string realDeviceName);
         code deregisterDevice(in string realDeviceName);
-        
+
         string getServantIOR(in string servantName);
     };
 };
-- 
2.41.0.windows.1

